std/json.d

```diff
@@ -346,10 +346,11 @@ struct JSONValue
             type_tag = JSON_TYPE.STRING;
             store.str = arg;
         }
-        else static if(is(T : char[]))
+        else static if(isNarrowString!T)
         {
+            import std.utf : toUTF8;
             type_tag = JSON_TYPE.STRING;
-            store.str = arg.idup;
+            store.str = toUTF8(arg);
```

Comment:
I think you need to use `byUTF!char.array`, which would replace invalid UTF sequences with replacement chars (instead of throwing `UTFException`), to allow `JSONValue.assign` to be callable from `nothrow` code.